core/src/main/scala/kafka/log/Log.scala

```diff
@@ -626,6 +618,24 @@ class Log(@volatile var dir: File,
         }
       }
     }
+
+    if (logSegments.nonEmpty) {
+      val logEndOffset = activeSegment.readNextOffset - 1
```

Comment:
Do you think it might slow startup if we have to read from every segment? I am wondering if we should leave the segments for deletion later and just use `max(activeSegment.readNextOffset, logStartOffset)` as the initial log end offset.

Reply:
My mistake. I read over this logic a bit too quickly.